managedtenants/utils/general_utils.py

```diff
@@ -23,3 +26,34 @@ def run(cmd, logger=None):
     return subprocess.run(
         cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=False
     )
+
+
+@contextmanager
+def timeout(duration):
+    def timeout_handler(signal, frame):
+        raise TimeoutError(f"Timed out after {duration} seconds")
+
+    signal.signal(signal.SIGALRM, timeout_handler)
+    signal.alarm(duration)
+    try:
+        yield
+    finally:
+        signal.alarm(0)
+
+
+def try_with_timeout_until(
```

Comment:
looks a lot like the sretoolbox.utils.retry https://github.com/app-sre/sretoolbox/blob/master/sretoolbox/utils/retry.py

Reply:
@apahim I'm not really sure the retry API provided in the sre toolbox is right here. The retry provided in the sre toolbox works on exceptions, whereas we want retry on predicates.

Reply:
fair enough